src/codeu/chat/server/View.java

```diff
@@ -181,6 +149,66 @@ public int conversationStatusUpdate(String title, Uuid owner) {
     return newMessages;
   }
 
+  private Collection<String> searchContributions(Time lastUpdate, Uuid searchUser) {
+
+    // Given a time value for the last time that a check for contributions was
+    // requested and the UUID of the user to search for recent contributions from,
+    // return a collection of the titles of conversations that the user has created
+    // or added messages to after the specified time.
+
+    Collection<String> contributions = new ArrayList<String>();
+    Collection<ConversationPayload> allConversations = all(model.conversationPayloadById());
+
+    // go through every conversation
+    for(ConversationPayload conversationPayload : allConversations) {
+      Message currentMessage = model.messageById().first(conversationPayload.firstMessage);
+      boolean foundMessage = false;
+      // go through every message
+      while(currentMessage != null && foundMessage == false) {
+        // check for a matching user UUID and a creation time after the last status update
+        if(lastUpdate.compareTo(currentMessage.creation) < 0 && currentMessage.author.equals(searchUser)) {
+          // add the conversation's title to the collection and break the loop for this conversation
+          contributions.add(model.conversationById().first(conversationPayload.id).title);
+          foundMessage = true;
+        }
+        currentMessage = model.messageById().first(currentMessage.next);
+      }
+      // check if the current conversation wasn't added  to the collection after the above loop
+      if(!contributions.contains(model.conversationById().first(conversationPayload.id).title)) {
+        // check if the current conversation's creator matches the specified user
```

Comment:
Also you're using `model.conversationById().first(conversationPayload.id)` few times. Can you assign it to a local variable to improve readability ?